kzg/src/eip_7594.rs

```diff
@@ -591,19 +619,50 @@ fn compute_r_powers_for_verify_cell_kzg_proof_batch<TG1: G1, TFr: Fr>(
 
 fn compute_weighted_sum_of_commitments<
     TG1: G1 + G1LinComb<TFr, TG1Fp, TG1Affine>,
-    TFr: Fr,
+    TFr: Fr + Send + Sync, //added + Send + Sync,
     TG1Fp: G1Fp,
     TG1Affine: G1Affine<TG1, TG1Fp>,
 >(
     commitments: &[TG1],
     commitment_indices: &[usize],
     r_powers: &[TFr],
 ) -> TG1 {
+    // let mut commitment_weights = vec![TFr::zero(); commitments.len()];
+
+    // for i in 0..r_powers.len() {
+    //     commitment_weights[commitment_indices[i]] =
+    //         commitment_weights[commitment_indices[i]].add(&r_powers[i]);
+    // }
+
     let mut commitment_weights = vec![TFr::zero(); commitments.len()];
 
-    for i in 0..r_powers.len() {
-        commitment_weights[commitment_indices[i]] =
-            commitment_weights[commitment_indices[i]].add(&r_powers[i]);
+    #[cfg(feature = "parallel")]
+    {
+        let intermediate_weights: Vec<_> = r_powers
+            .par_chunks(64) 
+            .zip(commitment_indices.par_chunks(64))
+            .map(|(r_chunk, idx_chunk)| {
+                let mut local_weights = vec![TFr::zero(); commitments.len()];
+                for (r_power, &index) in r_chunk.iter().zip(idx_chunk.iter()) {
+                    local_weights[index] = local_weights[index].add(r_power);
+                }
+                local_weights
+            })
+            .collect();
+
+        for local_weights in intermediate_weights {
+            for (i, weight) in local_weights.into_iter().enumerate() {
+                commitment_weights[i] = commitment_weights[i].add(&weight);
+            }
+        }
+    }
```

Comment:
Very clever! Have you tried picking chunk size, other than `64`? My intuition would be that picking `r_powers.len() / available_parallelism()` will show good results. Also, have you considered parallelizing summation part?